src/main/java/org/broadinstitute/hellbender/utils/clipping/ClippingOp.java

```diff
@@ -548,7 +545,8 @@ private CigarShift cleanHardClippedCigar(final Cigar cigar) {
                 } else if (!readHasStarted && cigarElement.getOperator() == CigarOperator.HARD_CLIP) {
                     totalHardClip += cigarElement.getLength();
                 } else if (!readHasStarted && cigarElement.getOperator() == CigarOperator.DELETION) {
-                    totalHardClip += cigarElement.getLength();
+                    //totalHardClip += cigarElement.getLength();
```

Comment:
It seems like you are putting in an effect-less line of code to document that the total is not incremented.  Is that because there are a bunch of `else if`s and you don't want to give the impression that it's always incremented?  How about this:
```
if (!readHasStarted && (cigarElement.getOperator() == CigarOperator.HARD_CLIP || cigarElement.getOperator() == CigarOperator.CigarOperator.SKIPPED_REGION) {
  totalHardClip += cigarElement.getLength();
}
readHasStarted |= cigarElement.getOperator() != CigarOperator.DELETION &&
                        cigarElement.getOperator() != CigarOperator.SKIPPED_REGION &&
                        cigarElement.getOperator() != CigarOperator.HARD_CLIP;
```

Also, in the (unmodified) code below it seems that the `if (ReadHasStartedBlock)` always results in `addedHardClips == true`.  If you agree, could you move the `addedHardCips = true` statements to the end of that block?